### ISSUE TITLE
Small error in the documentation for pb

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3054]
|<img alt="njal's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| njal reported|
|-|-|
|Trac ID|trac#3054|
|Type|defect|
|Created|2008-02-23 15:06:12Z|
|Branch|https://github.com/twisted/twisted/tree/setcopierforclass-docs-3054-3|

I was looking at the documentation for t.s.pb today (for Copyable specifically), and after following the pb-intro-howto, I found the following small issues:

http://twistedmatrix.com/trac/browser/trunk/doc/core/howto/pb-intro.xhtml#L243

should read 
```
flavors.setCopierForClass(Foo, RemoteFoo) # (or even setUnjellyableForClass)
```

As it stands now, using str(Foo), will result the follow if defined in module 'model':

```
twisted.spread.jelly.InsecureJelly: Module model not allowed (in type model.Foo).
```

I have not checked this with any other python-version than python 2.5.1 on OS X  Leopard.

Also, the docstring for this function: http://twistedmatrix.com/trac/browser/trunk/twisted/spread/jelly.py#L191
refers to MyCopier, but this probably should read MyJellier?

```
197	
198	        jellier.setUnjellyableForClass('module.package.Class', MyJellier).
199	
200	    Call this at the module level immediately after its class
201	    definition. MyCopier should be a subclass of RemoteCopy.
```

Attachments:

* [pbintro.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b64/b643584e5bbdabd7763e456caf4f043f4af1d29e/3a0d716e804c9cadf38b3142491004e1de4c1db9.patch) (1064 bytes) - added by njal on 2008-02-23 15:07:16Z - Diff against current trunk that changes the documentation and docstring.

<details><summary>Searchable metadata</summary>

```
trac-id__3054 3054
type__defect defect
reporter__njal njal
priority__normal normal
milestone__ 
branch__branches_setcopierforclass_docs_3054_3 branches/setcopierforclass-docs-3054-3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__pb pb
keywords__ 
time__1203779172000000 1203779172000000
changetime__1211889971000000 1211889971000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [22697]) Branching to 'setcopierforclass-docs-3054'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

I got rid of the rest of the uses of the setCopier APIs and gave pb.py an `__all__` too.  I deprecated some things too and did some other random junk.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

 * what's the purpose of _version1 in twisted/__init__.py?
 * pyflakes errors: ```twisted/enterprise/util.py:9: 'Version' imported but unused```
 * there are modifications in 2 untested files, gtkmanhole and spelunk_gnome. Tickets should probably be opened to add test for them

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [22728]) remove unused Version and unused import

refs [#3054](https://github.com/twisted/twisted/issues/3054)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [23246]) Branching to 'setcopierforclass-docs-3054-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [23708]) Branching to 'setcopierforclass-docs-3054-3'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun removed owner|
|-|-|

Forgot about this for a while.  Points from the review have been addressed, I think (I didn't file any tickets, I just dropped the changes in question).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set owner to @exarkun|
|-|-|

The deprecation should be updated to 8.2, now that 8.1 has been released. Otherwise, looks good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [23712]) Merge setcopierforclass-docs-3054-3

Author: exarkun
Reviewer: therve
Fixes: [#3054](https://github.com/twisted/twisted/issues/3054)

Update code and documentation which uses setCopierForClass to use setUnjellyableForClass,
which has been the preferred name for this API for many years.  Minor other improvements
to nearby code.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
